cmd/tcksample.cpp

```diff
@@ -186,11 +202,24 @@ class SamplerNonPrecise
     {
       out.set_index (tck.get_index());
       out.resize (tck.size());
-      for (size_t i = 0; i != tck.size(); ++i) {
-        if (interp.scanner (tck[i]))
-          out[i] = interp.value();
-        else
-          out[i] = NAN;
+      if (sh_precomputer) {
+        for (size_t i = 0; i != tck.size(); ++i) {
+          if (interp.scanner (tck[i])) {
+            for (interp.index(3) = 0; interp.index(3) != interp.size(3); ++interp.index(3))
+              sh_coeffs[interp.index(3)] = interp.value();
+            const Eigen::Vector3f dir = (tck[(i == ssize_t(tck.size()-1)) ? i : (i+1)] - tck[i ? (i-1) : 0]).normalized();
```

Comment:
Perhaps `std::clamp` conveys the intention slightly better here? Something like:
`const Eigen::Vector3f dir = (tck[std::clamp(i + 1, size_t(0), tck.size() - 1)] - tck[std::clamp(i - 1, size_t(0), tck.size() - 1)]).normalized();`

Reply:
I will need to reprogram my brain to be accustomed to having that one available. However the base for this branch is pre-`cmake` and so pre-C++17. 
(This particular calculation would also be better handled via #2189)